chips/apollo3/src/stimer.rs

```diff
@@ -0,0 +1,183 @@
+            HRFC_DIV16 = 0x1,
+            HRFC_DIV256 = 0x2,
+            XTAL_DIV1 = 0x3,
+            XTAL_DIV2 = 0x4,
+            XTAL_DIV32 = 0x5,
+            LFRC_DIV1 = 0x6,
+            CTIMER0A = 0x7,
+            CTIMER0B = 0x8
+        ],
+        COMPARE_A_EN OFFSET(8) NUMBITS(1) [],
+        COMPARE_B_EN OFFSET(9) NUMBITS(1) [],
+        COMPARE_C_EN OFFSET(10) NUMBITS(1) [],
+        COMPARE_D_EN OFFSET(11) NUMBITS(1) [],
+        COMPARE_E_EN OFFSET(12) NUMBITS(1) [],
+        COMPARE_F_EN OFFSET(13) NUMBITS(1) [],
+        COMPARE_G_EN OFFSET(14) NUMBITS(1) [],
+        COMPARE_H_EN OFFSET(15) NUMBITS(1) [],
+        CLEAR OFFSET(30) NUMBITS(1) [],
+        FREEZE OFFSET(31) NUMBITS(1) []
+    ],
+    STTMR [
+        STTMR OFFSET(0) NUMBITS(31) []
+    ],
+    CAPTURECONTROL [
+        CAPTURE0 OFFSET(0) NUMBITS(1) [],
+        CAPTURE1 OFFSET(1) NUMBITS(1) [],
+        CAPTURE2 OFFSET(2) NUMBITS(1) [],
+        CAPTURE3 OFFSET(3) NUMBITS(1) []
+    ],
+    SCMPR [
+        SCMPR OFFSET(0) NUMBITS(31) []
+    ],
+    SCAPT [
+        SCATP OFFSET(0) NUMBITS(31) []
+    ],
+    SNVR [
+        SNVR OFFSET(0) NUMBITS(31) []
+    ],
+    STMINT [
+        COMPAREA OFFSET(0) NUMBITS(1) [],
+        COMPAREB OFFSET(1) NUMBITS(1) [],
+        COMPAREC OFFSET(2) NUMBITS(1) [],
+        COMPARED OFFSET(3) NUMBITS(1) [],
+        COMPAREE OFFSET(4) NUMBITS(1) [],
+        COMPAREF OFFSET(5) NUMBITS(1) [],
+        COMPAREG OFFSET(6) NUMBITS(1) [],
+        COMPAREH OFFSET(7) NUMBITS(1) [],
+        OVERFLOW OFFSET(8) NUMBITS(1) [],
+        CAPTUREA OFFSET(9) NUMBITS(1) [],
+        CAPTUREB OFFSET(10) NUMBITS(1) [],
+        CAPTUREC OFFSET(11) NUMBITS(1) [],
+        CAPTURED OFFSET(12) NUMBITS(1) []
+    ]
+];
+
+pub struct STimer<'a> {
+    registers: StaticRef<STimerRegisters>,
+    client: OptionalCell<&'a dyn hil::time::AlarmClient>,
+}
+
+impl<'a> STimer<'_> {
+    const fn new(base: StaticRef<STimerRegisters>) -> STimer<'a> {
+        STimer {
+            registers: base,
+            client: OptionalCell::empty(),
+        }
+    }
+
+    pub fn handle_interrupt(&self) {
+        let regs = self.registers;
+
+        // Disable timer
+        regs.stcfg.modify(STCFG::COMPARE_A_EN::CLEAR);
+
+        // Disable interrupt
+        regs.stminten.modify(STMINT::COMPAREA::CLEAR);
+
+        // Clear interrupt
+        regs.stmintclr.modify(STMINT::COMPAREA::SET);
+
+        self.client.map(|client| client.fired());
+    }
+}
+
+impl<'a> hil::time::Alarm<'a> for STimer<'a> {
+    fn set_client(&self, client: &'a dyn hil::time::AlarmClient) {
+        self.client.set(client);
+    }
+
+    fn set_alarm(&self, tics: u32) {
+        let regs = self.registers;
+
+        // Set the clock source
+        regs.stcfg.write(STCFG::CLKSEL::XTAL_DIV2);
+
+        // Enable interrupts
+        regs.stminten.modify(STMINT::COMPAREA::SET);
+
+        // Set the delta
+        regs.scmpr[0].set(tics - regs.sttmr.get());
```

Comment:
This is a little different from every other `set_alarm()` implementation in Tock right now, because the compare register is set to a delta in the future rather than a precise tick count at which an interrupt will fire (Because the precise tick count is set in hardware). I dont think this is an issue, but may be worth considering as part of the time/alarm HIL redesign cc @phil-levis .